topi/python/topi/arm_cpu/bitserial_conv2d.py

```diff
@@ -1,331 +1,315 @@
+    assert pack_dtype == 'uint8', "only support packing into uint8 bits"
+    assert out_dtype == 'int16', "only support output type of int16"
 
-    data_q = bitpack(data, activation_bits, pack_axis=3, bit_axis=3, pack_type='uint8')
-    kernel_vec = _kernel_vec_spatial_pack_nhwc(kernel, weight_bits, VC)
-    N, H, W, IB, CI = data_q.shape
-    OCO, KH, KW, KB, VC, _ = kernel_vec.shape
+    N, H, W, CI = get_const_tuple(data.shape)
+    if len(kernel.shape) == 4:
+        KH, KW, _, CO = get_const_tuple(kernel.shape)
+        CI_packed = CI // 8
+    else:
+        KH, KW, KB, CI_packed, CO = get_const_tuple(kernel.shape)
 
-    CO = OCO * VC
-    HPAD, WPAD, _, _ = get_pad_tuple(padding, kernel)
+    if isinstance(padding, int) or (isinstance(padding, (tuple, list)) and len(padding) == 2):
+        TPAD, LPAD, DPAD, RPAD = get_pad_tuple(padding, kernel)
+    else:
+        TPAD, LPAD, DPAD, RPAD = padding
 
     if isinstance(stride, (tuple, list)):
         HSTR, WSTR = stride
     else:
         HSTR, WSTR = stride, stride
     HCAT, WCAT = KH-1, KW-1
 
-    PAD_H = H + 2*HPAD
-    PAD_W = W + 2*WPAD
-    OH = (H + 2*HPAD - KH) // HSTR + 1
-    OW = (W + 2*WPAD - KW) // WSTR + 1
+    PAD_H = H + (TPAD + DPAD)
+    PAD_W = W + (LPAD + RPAD)
+    OH = (PAD_H - KH) // HSTR + 1
+    OW = (PAD_W - KW) // WSTR + 1
+    oshape = (1, OH, OW, CO)
+
+    # Pad input channels of weights and data when it is not a multiple of 8
+    if CI_packed % 8 != 0:
+        CI_PAD = CI_packed % 8
+        CI_packed += CI_PAD
+    else:
+        CI_PAD = 0
+
+    # ==================== define configuration space ====================
+    n, oh, ow, co = cfg.axis(N), cfg.axis(OH), cfg.axis(OW), cfg.axis(CO)
+    ci, kh, kw = cfg.reduce_axis(CI_packed), cfg.reduce_axis(KH), cfg.reduce_axis(KW)
+    ib, kb = cfg.reduce_axis(activation_bits), cfg.reduce_axis(weight_bits)
+
+    co, vc = cfg.define_split('tile_co', co, policy='all', num_outputs=2,
+                              filter=lambda x: x.size[-1] == 8)
+    oh, vh = cfg.define_split('tile_oh', oh, policy='all', num_outputs=2,
+                              filter=lambda x: x.size[-1] >= 2)
+    ow, vw = cfg.define_split('tile_ow', ow, policy='all', num_outputs=2,
+                              filter=lambda x: x.size[-1] >= 2)
+    # cfg.define_annotate('ann_reduce', [ib, kb, kh, kw], policy='try_unroll')
+    ci_o, ci_i = cfg.define_split("tile_ci", ci, num_outputs=2,
+                                  filter=lambda x: x.size[-1] == 8 or x.size[-1] == 16)
+    re_axes = cfg.define_reorder("reorder_0",
+                                 [n, oh, ow, co, vh, vw, kh, kw, ci_o, kb, ib, vc, ci_i],
+                                 policy='candidate', candidate=[
+                                     [n, oh, ow, co, vh, vw, kh, kw, ci_o, kb, ib, vc, ci_i],
+                                     [n, oh, ow, co, vh, vw, kw, kh, ci_o, kb, ib, vc, ci_i],])
+    cfg.add_flop(2 * N * OH * OW * CO * CI * 8 * KH * KW)
```

Comment:
Does the automatic flop calculation in #2776 work for this?

Reply:
Sadly no :( it gives an error about multiple output in the compute.